pkg/datapath/tables/node_address_test.go

```diff
@@ -141,42 +141,6 @@ var nodeAddressTests = []struct {
 			netip.MustParseAddr("2001:db8::1"),
 		},
 	},
-	{
```

Comment:
Having to delete a test means that we're changing a behaviour chosen and specified consciously; so I'm a bit wary of this change having a larger impact than intended. You're interested in fixing something for GKE but changing the behaviour for all platforms. 

cc @joamaki for the devices/node addr knowledge.

Reply:
btw Jussi is on the PR https://github.com/cilium/cilium/pull/41645, and I was asked to create this PR to seperate this commit after his approval.